### PR TITLE
Feat/359 add tokens to mm

### LINKED ISSUE #359: Add vega token to wallet (like in SushiSwap)
screenshot_2021-09-03_at_14 30 05_720

There is a way to have a button in you your ui to prompt a message in metamask to ask if you want to add a token.

This to be added instead of text prompt.

```
const tokenAddress = '0xd00981105e61274c8a5cd5a88fe7e037d935b513';
const tokenSymbol = 'TUT';
const tokenDecimals = 18;
const tokenImage = 'http://placekitten.com/200/300';

try {
  // wasAdded is a boolean. Like any RPC method, an error may be thrown.
  const wasAdded = await ethereum.request({
    method: 'wallet_watchAsset',
    params: {
      type: 'ERC20', // Initially only supports ERC20, but eventually more!
      options: {
        address: tokenAddress, // The address that the token is at.
        symbol: tokenSymbol, // A ticker symbol or shorthand, up to 5 chars.
        decimals: tokenDecimals, // The number of decimals in the token
        image: tokenImage, // A string url of the token logo
      },
    },
  });

  if (wasAdded) {
    console.log('Thanks for your interest!');
  } else {
    console.log('Your loss!');
  }
} catch (error) {
  console.log(error);
}
```
AND there's a flag so we can check and only show on meta mask users browsers :exploding_head:
`window.ethereum.isMetaMask`

### PR DESCRIPTION
**Pull request by dexturr**
Closes #359

Adds env after the asset you are adding (unless on mainnet):

Screenshot 2021-11-12 at 10 39 27

Prompts after claim:

Screenshot 2021-11-12 at 10 39 07

Prompts on vesting page: 
Screenshot 2021-12-13 at 10 03 39

### CODE REVIEW

**Comment by mattrussell36** [comment]
@dexturr Whats the status on this can we rebase and get it merged?

File: src/components/add-token-button/add-token-button.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    symbol,
+    decimals,
+    image
+  );
+  if (!addSupported) {
+    return null;
+  }
+  return (
+    <button className="add-token-button button-link" onClick={add}>
+      Click here to add to wallet
```
Comment (campbellssource) [inline src/components/add-token-button/add-token-button.tsx:25]: Could we make this a translation.
and make the copy = "Show this token in your Ethereum wallet"

**Review by campbellssource — DISMISSED** [review]

**Comment by campbellssource** [comment]
LGTM 👍

File: src/components/add-locked-token/add-locked-token.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.or {
```
Comment (mattrussell36) [inline src/components/add-locked-token/add-locked-token.scss:1]: Optional nitfpick: We should probably name space this properly and perhaps give it a slightly better name, perhaps: `.add-locked-token-address__or-divider`

File: src/components/vega-wallet/vega-wallet.tsx [hunk 3]
```diff
@@ -440,7 +445,8 @@ const VegaWalletConnected = ({
         image={vegaWhite}
         decimals={decimals}
         name="VEGA"
-        symbol="associated"
+        subheading={"associated"}
```
Comment (mattrussell36) [inline src/components/vega-wallet/vega-wallet.tsx:448]: Should this be a translation string?

File: src/config/ethereum.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { Networks } from "./vega";
 
 const appChainId = process.env.REACT_APP_CHAIN as EthereumChainId;
-const appEnv = process.env.REACT_APP_ENV as Networks;
+export const appEnv = process.env.REACT_APP_ENV as Networks;
```
Comment (mattrussell36) [inline src/config/ethereum.ts:4]: Perhaps this pattern is unnecessary but later on in this file we export `appChainId` as `APP_CHAIN_ID`. Might be good to be consistent and export `appEnv` as `APP_ENV` so we follow the 'best practice' of upper camel case for constants

Comment (mattrussell36) [reply src/config/ethereum.ts:4]: Or just define them immediately as upper camel case

File: src/hooks/use-add-asset-to-wallet.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        params: {
+          type: "ERC20",
+          options: {
+            address,
+            symbol: `${symbol}${
+              // Add the environment if not mainnet
+              appEnv === Networks.MAINNET
+                ? ""
+                : // Remove NET as VEGA(TESTNET) is too long
+                  ` ${appEnv.replace("NET", "")}`
```
Comment (mattrussell36) [inline src/hooks/use-add-asset-to-wallet.ts:42]: Kind of feel like this should be VEGA(TEST) rather than VEGA(NET)? If it fits?

**Review by mattrussell36 — APPROVED** [review]

File: src/hooks/use-add-asset-to-wallet.ts (hunk 5 above)
Comment (dexturr) [reply src/hooks/use-add-asset-to-wallet.ts:42]: That's what this does, we are removing the NET part so:
* stagnet -> stag
* testnet -> test
* stagnet2 -> stag2
* mainnet -> ignored

**Pull request merged**